TOOLS/osxbundle.py

```diff
@@ -77,7 +78,8 @@ def main():
 
     if options.deps:
         print("> bundling dependencies")
-        print(sh(" ".join([os.path.join(src_path, "TOOLS/dylib-unhell.py"), target_binary(binary_name)])))
+        cmd = " ".join([os.path.join(src_path, "TOOLS/dylib-unhell.py"), target_binary(binary_name)])
+        subprocess.run(cmd, shell=True, universal_newlines=True, check=True)
```

Comment:
I don't think you need shell for this, do you? Just run python directly and don't splice command in one string, pass the args array.

Reply:
i am assuming you mean running the script in a subprocess not the `shell=True`? for the former, i suggested that on previous occasions and it was decided to leave it as is. reason was manly performance improvements. anyway i don't want to go against that decision just like this and possible piss people of. hence why i left the subprocess.
if you actually meant the latter, i thought popen (which was used previously with the `sh` function) had `shell=True` as default. that is my bad, i got it wrong.

`shell=True is also the actual reason why i left it as a string and didn't change it to a args sequence, from the python docs:
```
 If shell is True, it is recommended to pass args as a string rather than as a sequence.
```

we can revisit that decision, though i don't think we should do it on this PR.

Reply:
on a side note, i will adjust the script a bit and will make a small performance comparison. maybe the performance point is moot by now and we could just get rid of it.

Reply:
I just meant the `shell=True` part, if we only want to run a script (like we do) simple subprocess is enough without circling back through shell command parsing and other things that happen there.

Reply:
thanks for the heads up. i changed it to directly using the script now. made some performance comparisons and there are no differences whatsoever.

```
with import:
16.932
16.656
16.582
16.834
16.667
16.728
16.576
16.693
16.721
16.562
Average: 16.6951

with subprocess:
16.616
16.668
16.798
16.671
16.648
16.632
16.633
16.621
16.578
16.643
Average: 16.6508
``